src/server/Parser/liquidTagGrammar.ne

```diff
@@ -4,16 +4,62 @@
 
+# For the input:
+# {% entityform id:'', name:"xyz",  key:"34234" %}
+# The output of the parser is generated in the following format:
+
+# token: Array
+# output: Object
+#	tag: "entityform"
+#	map: Object
+#     18: "id"
+#     27: "name"
+#     39: "key"
+
+# How to access the ouptut:-
+# parser.results[0]?.output?.tag
+# parser.results[0]?.output?.map
+
 @builtin "whitespace.ne" # `_` means arbitrary amount of whitespace
 @builtin "number.ne"     # `int`, `decimal`, and `percentage
 @builtin "string.ne"     # "strings"
-LiquidExpression -> _ "{%" __ TAG_DEFINITION __ "%}" _ {% function(token) {return { token:token, output: {tag: token[3].tag, key: token[3].key, value: token[3].value}}} %}
-TAG_DEFINITION -> TAG __ ATTRIBUTE_MAP {% function(token) {return { token:token, tag:token[0].output, key:token[2].key, value:token[2].value }} %}
-TAG -> ("entityform" | "webform" | "entityview" | EntityList) {% function(token) {return { output:token[0]}} %}
-EntityList -> LIQUID_KEYWORD _ ENTITYLIST_TAG {% function(token) {return { output:token[2]}} %}
-LIQUID_KEYWORD -> "include" {% function(token) {return { output:token[0]}} %}
-ENTITYLIST_TAG -> "'entity_list'" {% function(token) {return { output:"entityList"}} %}
-KEY -> ("id" | "name" | "key" ) {% id %}
-VALUE -> sqstring | dqstring {% id %}
-ATTRIBUTE_MAP -> PAIR {% function(token) {return { key: token[0].key, value: token[0].value}} %}
-PAIR -> KEY _ ":" _ VALUE {% function(token) {return { key:token[0], value: token[4]}} %}
+LiquidExpression -> _ "{%" __ TAG_DEFINITION __ "%}" _ {% function(token) {return { token:token, output: {tag: token[3].tag, map: token[3].map}}} %}
+TAG_DEFINITION -> TAG __ ATTRIBUTE_MAP {% function(token) {return { token:token, tag:token[0].tag, map: token[2] }} %}
+TAG -> "entityform"  {% function(token) {return { tag: token[0] }} %}
+		| "webform"  {% function(token) {return { tag: token[0]}} %}
+		| "entityview" {% function(token) {return { tag: token[0]}} %}
+		| EntityList {% function(token) {return { tag: token[0].tag }} %}
+EntityList -> LIQUID_KEYWORD _ ENTITYLIST_TAG {% function(token) {return { tag:token[2].tag}} %}
+LIQUID_KEYWORD -> "include" {% id %}
+ENTITYLIST_TAG -> "'entity_list'" {% function(token) {return { tag: "entityList" }} %}
+ATTRIBUTE_MAP -> _ (PAIR _):+  {% extractObjectFromSpaceSeparatedPairs %}
+	            | (_ PAIR _ ","):+ _ PAIR _ {% extractObjectFromCommaSeparatedPairs %}
+PAIR -> KEY _ ":" _ VALUE {% function(token) { return [token[0], token[4]]; } %}
+KEY -> ("id" | "name" | "key") {% id %}
+VALUE -> sqstring {% function(token, loc) {return { value: token[0], location: loc}} %}
+		| dqstring {% function(token, loc) {return { value: token[0], location: loc }} %}
+
+@{%
+
+function extractPair(kv, output) {
+    if(kv[0]) { output[kv[1].location + 1] = kv[0][0]; } // kv[0] is key and kv[1] is value
+}
+
+function extractObjectFromSpaceSeparatedPairs(d) {
```

Comment:
not yet blocking for this PR, but this grammar is getting (as expected) more complex, and the helper functions need to reach into non-obvious match structures to extract the relevant info. -> unit tests will help future regressions

Reply:
Yes David, post this release we will start focusing on the Unit Tests.